### DISCUSSION TITLE
venus v1.2.0 Release Timeline about the Network V15 OhSnap upgrade

### DISCUSSION THREAD

**Discussion by Joss-Hua** [filecoin-project/venus#4688] [Announcements]
**More details on the OhSnap network upgrade can be found [here](https://github.com/filecoin-project/community/discussions/74#discussioncomment-1922550)!**

If you have any concerns or questions, leave a comment [here](https://github.com/filecoin-project/venus/discussions/4689).

## Venus release
**Note: Venus will be released no later than 24 hours after the release of Lotus. This projected timeline may be subject to change based on testings and developments!**

You will need to be on venus v1.2.0 or ^ for the OhSnap upgrade. 

- [venus `v1.2.0`](https://github.com/filecoin-project/venus/releases/tag/v1.2.0): It is based on the last release, venus v1.1.3-rc1, and includes consensus-breaking changes and a couple of necessary improvements for the Network v15 OhSnap upgrade. 
   - Ecosystem teams such as node operators and exchanges need to upgrade your node to this version immediately after the release of the tag, and only need to run the chain service modules; After the release of this version, users such as storage providers and data agents need to upgrade your nodes and all systems to this version!

As mentioned in the [OhSnap Notice](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-1927638):

- venus release candidates will include the temporary proof params for the community to test Snap Deal in the SnapNet.
   - Note: the temporary proof params are not available in master to prevent users from using the wrong params in mainnet.
- Final releases will be out after the trusted setup ceremony is completed and the final proof is available. New release candidates will be published before the calibration upgrade once the final proof is ready.

## Venus release timeline
> Time below is in **UTC**.

Please pay attention to the changes.
- [x] 2022-01-13, [venus v1.2.0-rc2](https://github.com/filecoin-project/venus/releases/tag/v1.2.0-rc2) (Note: This release candidate uses temporary proof params for Snap Deals and only sets upgrade epcoh for Butterfly-SnapNet, does not set the upgrade epochs for calibration and mainnet.)
- [x] 2022-01-13, [venus-wallet v1.4.0-rc1](https://github.com/filecoin-project/venus-wallet/releases/tag/v1.4.0-rc1), [venus-gateway v1.2.0-rc1](https://github.com/ipfs-force-community/venus-gateway/releases/tag/v1.2.0-rc1), [venus-messager v1.4.0-rc1](https://github.com/filecoin-project/venus-messager/releases/tag/v1.4.0-rc1), [venus-sealer v1.4.0-rc2](https://github.com/filecoin-project/venus-sealer/releases/tag/v1.4.0-rc2), [venus-miner v1.4.0-rc2](https://github.com/filecoin-project/venus-miner/releases/tag/v1.4.0-rc2), [venus-market v2.0.0-rc2](https://github.com/filecoin-project/venus-market/releases/tag/v2.0.0-rc2)
- [x] 2022-01-17T19:00:00Z, Butterfly-SnapNet upgrade @epoch 30262 !
- [x] 2022-01-18, [venus v1.2.0-rc3](https://github.com/filecoin-project/venus/releases/tag/v1.2.0-rc3), fix sync issues
- [x]  2022-01-19, [venus v1.2.0-rc4](https://github.com/filecoin-project/venus/releases/tag/v1.2.0-rc4), fix sync issues
- [x] 2022-01-26,  [venus-miner v1.4.0-rc3](https://github.com/filecoin-project/venus-miner/releases/tag/v1.4.0-rc3), trusted setup ceremony ends
- [x] 2022-01-28, [venus-sealer v1.4.0-rc3](https://github.com/filecoin-project/venus-sealer/tree/v1.4.0-rc3), snapdeal supporting
- [x] 2022-02-08, release candidates with final proof parameters
- [x] 2022-02-08, [venus v1.2.0-rc5](https://github.com/filecoin-project/venus/releases/tag/v1.2.0-rc5), reset butterfly network
- [x] 2022-02-10T19:23:00Z, calibration upgrade @epoch 682006.
- [x] 2022-02-11, [venus-miner v1.4.0-rc4](https://github.com/filecoin-project/venus-miner/releases/tag/v1.4.0-rc4), [venus-sealer v1.4.0-rc5](https://github.com/filecoin-project/venus-sealer/releases/tag/v1.4.0-rc5)
- [x] ~2022-02-18~ 2022-02-21, [venus v1.2.0](https://github.com/filecoin-project/venus/releases/tag/v1.2.0)
- [x] ~2022-02-18~ 2022-02-21, [chain-co](https://github.com/ipfs-force-community/chain-co/releases/tag/v0.1.0) : add interface, 'MpoolPending', 'MpoolGetNonce'
- [x] ~2022-02-18~ 2022-02-21, [venus-wallet v1.4.0](https://github.com/filecoin-project/venus-wallet/releases/tag/v1.4.0), [venus-gateway v1.2.0](https://github.com/ipfs-force-community/venus-gateway/releases/tag/v1.2.0), [venus-messager v1.4.0](https://github.com/filecoin-project/venus-messager/releases/tag/v1.4.0), [venus-sealer v1.4.0](https://github.com/filecoin-project/venus-sealer/releases/tag/v1.4.0), [venus-miner v1.4.0](https://github.com/filecoin-project/venus-miner/releases/tag/v1.4.0), [venus-market v2.0.0](https://github.com/filecoin-project/venus-market/releases/tag/v2.0.0)
- [x] 2022-02-24  [venus v1.2.1](https://github.com/filecoin-project/venus/releases/tag/v1.2.1) [venus-sealer v1.4.1](https://github.com/filecoin-project/venus-sealer/releases/tag/v1.4.1)
- [x] 2022-03-01  [venus v1.2.2](https://github.com/filecoin-project/venus/releases/tag/v1.2.2)
- [x] 2022-03-01, mainnet upgrade

**Comment by cloudxin** [comment]
All components of **Calibnet** of the incubator will be updated at **15:00 p.m.** Beijing time on **February 10**. Users accessing this test network should pay attention to updating the corresponding versions of **venus-sealer** and **venus-wallet** in time

**Comment by cloudxin** [comment]
This upgrade needs to download an additional 32G file on the proof parameter file of Venus sealer host. Chinese users can obtain the new version of venus-sealer and IPFS_GATEWAY, and users in other regions can obtain it directly by using venus-sealer.

```
cd venus-sealer
git pull
git checkout v1.4.0-rc5
make clean && make deps && make 
```

_China region_:
**IPFS_GATEWAY="https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/" ./venus-sealer fetch-params 32GiB**

_Other regions_:
**./venus-sealer fetch-params 32GiB**

**Reply by cloudxin** [reply]
To obtain the proof parameter file, you can also use the Venus process of the specified version to obtain:
**./venus  fetch --size=34359738368**

**Reply by Imccccc** [reply]
how can I download the proofs file in specific path? Not enough space for /root.

**Reply by Fatman13** [reply]
@Imccccc 呦，学弟你好。👋 我也是普渡的，有兴趣私聊交流下？我在slack上是@fatman13。

**Comment by cloudxin** [comment]
If you upgrade to the V15 network venus-sealer version, the **venus-worker** running the C2 process also needs 32GiB **v28-empty-sector-update-merkletree-poseidon_hasher-8-8-0-3b7f44a9362e3985369454947bc94022e118211e49fd672d52bec1cbfd599d18.params** proof parameter files

**Comment by simlecode** [comment]
venus v1.2.2
lotus for venuspool venus/v1.14.0
chain-co v0.1.0
venus-gateway v1.2.0
venus-messager v1.4.0
wallet v1.4.0
venus-market v2.0.0
venus-sealer v1.4.1
venus-miner v1.4.0
